contracts/protocols/polygon/mstable/main.sol

```diff
@@ -0,0 +1,169 @@
+        //
+        if (_input == mUsdToken) {
+            // Check if mUSD
+            // mUSD is 1:1, doesn't need to be minted
+            return _amount;
+        } else if (IMasset(mUsdToken).bAssetIndexes(_input) != 0) {
+            // Check if bAsset
+            // Estimate mint
+            return IMasset(mUsdToken).getMintOutput(_input, _amount);
+        } else {
+            // Supplied token is neither mUSD nor bAsset, use function to supply a feeder pool address
+            revert("Token is not a bAsset or mUSD");
+        }
+    }
+
+    /**
+     * @dev Estimates deposits via Feeder Pool
+     * @notice Estimates the minted mUSD amount via Feeder Pool
+     * @param _input The input token to estimate
+     * @param _amount Amount of bAssets (or mUSD) to be deposited
+     * @param _path address of the Feeder Pool
+     * @return estimation of output
+     */
+    function estimateDeposit(
+        address _input,
+        uint256 _amount,
+        address _path
+    ) public view returns (uint256) {
+        return IFeederPool(_path).getSwapOutput(_input, mUsdToken, _amount);
+    }
+
+    /**
+     * @dev Estimates withdrawals to mUSD or bAsset
+     * @notice Estimates the output, with mUSD is 1:1, bAssets are calculated
+     * @param _output The output token to estimate
+     * @param _amount Amount of bAssets (or mUSD) to be withdrawn
+     * @return estimation of output
+     */
+    function estimateWithdrawal(address _output, uint256 _amount) public view returns (uint256) {
+        //
+        if (_output == mUsdToken) {
+            // Check if mUSD
+            // mUSD is 1:1, doesn't need to be minted
+            return _amount;
+        } else if (IMasset(mUsdToken).bAssetIndexes(_output) != 0) {
+            // Check if bAsset
+            // Estimate mint
+            return IMasset(mUsdToken).getRedeemOutput(_output, _amount);
+        } else {
+            // Supplied token is neither mUSD nor bAsset, use function to supply a feeder pool address
+            revert("Token is not a bAsset or mUSD");
+        }
+    }
+
+    /**
+     * @dev Estimates withdrawals via Feeder Pool
+     * @notice Estimates the output via Feeder Pool
+     * @param _output The output token to estimate
+     * @param _amount Amount to be withdrawn
+     * @param _path address of the Feeder Pool
+     * @return estimation of output
+     */
+    function estimateWithdrawal(
+        address _output,
+        uint256 _amount,
+        address _path
+    ) public view returns (uint256) {
+        return IFeederPool(_path).getSwapOutput(mUsdToken, _output, _amount);
+    }
+
+    /**
+     * @dev Retrieves Reward amounts
+     * @notice Rewards are two tokens
+     * @param _account address of the account to retrieve Reward data from
+     * @return rewards => MTA
+     * @return platformRewards => wMATIC
+
+     */
+    function getRewards(address _account) public view returns (uint256, uint256) {
```

Comment:
Also, I think it would be better to move functions with names starting from `get` into `helper.sol`